scripts/asm-installer/install_asm

```diff
@@ -1804,25 +1804,26 @@ EOF
 required_iam_roles() {
   # meshconfig.admin - required for init, stackdriver, UI elements, etc.
   # servicemanagement.admin/serviceusage.serviceUsageAdmin - enables APIs
-  if can_modify_gcp_components || \
+  if should_validate || \
```

Comment:
`should_validate` will be true most of the time, so this practically gets rid of the conditions for the IAM roles.

What we really want to check is `[[ "${ONLY_VALIDATE}" -eq 1 ]] && [[ "${ENABLE_ALL}" -eq 1 || "${ENABLE_FOO}" -eq 1`. I can't really think of a decent way to do that cleanly.

Reply:
Oh right, I meant to check `ONLY_VALIDATE` case. I create this PR to show a potential direction to fix. I agree that it should be easier after refactoring. Closing this PR.